cli/main.c

```diff
@@ -1451,11 +1447,12 @@ static int fw_update(int argc, char **argv)
 		}
 	}
 
-	ret = compare_device_secure_version(cfg.dev, fileno(cfg.fimg),
-					    cfg.assume_yes);
-	if (ret) {
-		fclose(cfg.fimg);
-		return ret;
+	if (!cfg.assume_yes && !switchtec_is_gen3(cfg.dev)) {
```

Comment:
I'd still keep the switchtec_is_gen3() inside the static function....

And in fact it sure looks like this could be a useful library function if restructured properly.... It's much better in the long run to add simple interfaces like that do complicated things based on internal state. Otherwise people will just copy the non-library code into their project and you break their project if you ever want to change it.

```
ret = switchtec_secure_version_matches_file(cfg.dev, fileno(cfg.fimg));
if (!ret) {
        /* print warning */
        ret = ask_if_sure(cfg.assume_yes);
        if (ret) {
                fclose(cfg.fimg);
                return ret;
        }
}
```

Reply:
The warning message needs to contain _device_secure_version_ and _file_secure_version_. The function needs to return these fields. The code looks like:
`ret = switchtec_secure_version_matches_file(cfg.dev, fileno(cfg.fimg), &dev_sec_ver, &file_sec_ver);
if (!ret) {
        printf("Warning: updating this file updates device secure version from %x to %x\n", dev_sec_ver, file_sec_ver);
        ret = ask_if_sure(cfg.assume_yes);
        if (ret) {
                fclose(cfg.fimg);
                return ret;
        }
}`

Reply:
Then two functions that return the sec_ver for a device and a file would be more appropriate...

Reply:
Function is now named "switchtec_fw_file_secure_version_newer()".